### PR TITLE
Homework3 part2

### PR DESCRIPTION
**Pull request by citikot**
Greetings!

Fixes applied:
1. common field name moved to parent class Animal.
2. Getter and setter removed from subclasses and created in parent class.
3. equal and hashcode removed from subclasses and created in parent class.
4. Method "swim" remains without changes according instructions of Sergey during the lesson. He admitted that conditions in the LMS are little bit incorrect and advised to create "swim" in subcluss.
5. File name of TestCase changed.

### CODE REVIEW

File: homeworks/src/homework3/part2/Animal.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        animalsCounter += 1;
+    }
+
+    public int getAnimalsCounter(){
+        return animalsCounter;
+    }
+
+    public void run(String name, int distance){
+        System.out.println(name + " пробежал " + distance + " м");
+    }
```
Comment (mkharkhuta) [inline homeworks/src/homework3/part2/Animal.java:18]: the name field is redundant, it is coming from Animal class

File: homeworks/src/homework3/part2/Cat.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        setName(name);
+    }
+
+    public int getCatsCounter() {
+        return catsCounter;
+    }
+
+    public void run(int distance) {
+        if (distance <= CAT_RUN_LIMIT) {
+            super.run(getName(), distance);
```
Comment (mkharkhuta) [inline homeworks/src/homework3/part2/Cat.java:24]: getName() is redundant

File: homeworks/src/homework3/part2/Dog.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        setName(name);
+    }
+
+    public int getDogsCounter() {
+        return dogsCounter;
+    }
+
+    public void run(int distance) {
+        if (distance <= DOG_RUN_LIMIT) {
+            super.run(getName(), distance);
```
Comment (mkharkhuta) [inline homeworks/src/homework3/part2/Dog.java:25]: the same

File: homeworks/src/homework3/part2/Animal.java (hunk 1 above)
Comment (citikot) [reply homeworks/src/homework3/part2/Animal.java:18]: got it! fixed.

File: homeworks/src/homework3/part2/Cat.java (hunk 2 above)
Comment (citikot) [reply homeworks/src/homework3/part2/Cat.java:24]: fixed

File: homeworks/src/homework3/part2/Dog.java (hunk 3 above)
Comment (citikot) [reply homeworks/src/homework3/part2/Dog.java:25]: fixed

**Comment by citikot** [comment]
finally got the idea. fixes by next PR.

**Pull request closed without merging**